### PR TITLE
Add "Same price by default for shops and stalls" interface option

### PR DESCRIPTION
**Pull request by rbndelrio**
Hey guys! I've got a QoL feature-request-turned-PR:

This is a quality of life feature intended to compliment the "Auto open shops" setting under Interface Tweaks. It automatically checks "Same price throughout park" for newly created shops and stalls.

image

For predictability/compatibility, it only functions under these conditions:

1. Ride object must be a shop or stall (excludes ride photos).
2. `ShopItem` shouldn't already have the "Same Price" flag
3. The shop/stall shouldn't already exist in the active park. This is to prevent overriding shop prices in saved games and new scenarios.
4. The item itself should also not exist in the park. For example, when placing Info Kiosk 1 with a Souvenir Shop present, the Park Map will have "Same Price" checked while the Umbrella will remain unchecked.

image

In practice, this setting combined with "Auto open shops" enables rapid placement of shops & stalls in the early game and reduces the overhead of micromanaging later on.

This being my first time in this codebase (and C++ lol) there's probably a better implementation out there. Most of this logic occurs inside the `RideCreateAction` and `ShopItem` but maybe it should take place in `RideConstructionWindow` where Auto Open Stalls and default inspection intervals happen?

Also as it stands this will conflict with flag updates in #17661 until it's merged.

Thanks!

### CODE REVIEW

**Comment by Gymnasiast** [comment]
I don’t think we should pile on another option when at least 99% of users either doesn’t care about this or would want the feature to be on all the time. Unnecessary options just clutter up the window and code base.

File: src/openrct2/ride/ShopItem.cpp [hunk 1]
```diff
@@ -149,6 +149,33 @@ bool shop_item_has_common_price(const ShopItem shopItem)
...
+{
+    if (shopItem == ShopItem::None)
+        return false;
+
+    for (const auto& ride : GetRideManager())
+    {
+        if (ride.id == excludeRideId)
+            continue;
+
+        if (shopItem == ShopItem::Admission && ride.type == RIDE_TYPE_TOILETS)
```
Comment (frutiemax) [inline src/openrct2/ride/ShopItem.cpp:162]: Don't add any RIDE_TYPE_* reference, we are currently trying to remove these from the code base.

File: src/openrct2/ride/ShopItem.cpp [hunk 2]
```diff
@@ -149,6 +149,33 @@ bool shop_item_has_common_price(const ShopItem shopItem)
...
+            return true;
+
+        const auto rideEntry = ride.GetRideEntry();
+        if (rideEntry == nullptr)
+            continue;
+
+        for (auto currentShopItem : rideEntry->shop_item)
```
Comment (frutiemax) [inline src/openrct2/ride/ShopItem.cpp:169]: Use reference

File: src/openrct2/ride/ShopItem.h [hunk 3]
```diff
@@ -132,5 +132,6 @@ extern uint64_t gSamePriceThroughoutPark;
 
 money32 shop_item_get_common_price(Ride* forRide, const ShopItem shopItem);
 bool shop_item_has_common_price(const ShopItem shopItem);
+bool shop_item_sold_elsewhere(const ShopItem shopItem, RideId excludeRideId);
```
Comment (frutiemax) [inline src/openrct2/ride/ShopItem.h:135]: Use PascalCase for new functions, also use const reference to avoid copy.

**Comment by frutiemax** [comment]
I'm also not too sure how the Default Inspection interval is working from your code, it just does not seem implemented... The usefulness of this PR is also very niche I'm afraid.

I'd rather see a plugin that does the job of checking that property for every shop. This way you avoid a PR and you can customize that however you want it.

**Comment by Gymnasiast** [comment]
Closing as it appears to be abandoned.

**Pull request closed without merging**